ryhope/src/storage/memory.rs

```diff
@@ -411,86 +411,89 @@ impl EpochMapItem {
+            ) => first_user_epoch.cmp(second_user_epoch),
             (
                 EpochMapItem::PartialIncremental(first_incremental_epoch),
                 EpochMapItem::Complete(_, second_incremental_epoch),
-            ) => Some(first_incremental_epoch.cmp(&second_incremental_epoch)),
+            ) => first_incremental_epoch.cmp(second_incremental_epoch),
             (
                 EpochMapItem::Complete(first_user_epoch, _),
                 EpochMapItem::PartialUser(second_user_epoch),
-            ) => Some(first_user_epoch.cmp(&second_user_epoch)),
+            ) => first_user_epoch.cmp(second_user_epoch),
             (
                 EpochMapItem::Complete(_, first_incremental_epoch),
                 EpochMapItem::PartialIncremental(second_incremental_epoch),
-            ) => Some(first_incremental_epoch.cmp(&second_incremental_epoch)),
+            ) => first_incremental_epoch.cmp(second_incremental_epoch),
             (
                 EpochMapItem::Complete(first_user_epoch, first_incremental_epoch),
                 EpochMapItem::Complete(second_user_epoch, second_incremental_epoch),
             ) => {
-                let user_epoch_cmp = first_user_epoch.cmp(&second_user_epoch);
-                let incremental_epoch_cmp = first_incremental_epoch.cmp(&second_incremental_epoch);
+                let user_epoch_cmp = first_user_epoch.cmp(second_user_epoch);
+                let incremental_epoch_cmp = first_incremental_epoch.cmp(second_incremental_epoch);
                 assert_eq!(
                     user_epoch_cmp, incremental_epoch_cmp,
                     "Breaking invariant of `EpochMapper`: both `UserEpoch` and `IncrementalEpoch`
                     must be monotonically increasing"
                 );
-                Some(user_epoch_cmp)
+                user_epoch_cmp
             }
             _ =>
             // all other cases are partial `EpochMapItem`s, which are never compared
             {
-                None
+                unreachable!()
             }
         }
     }
 }
 
-impl Ord for EpochMapItem {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        // Safe since by construction we will never compare 2 Partial
-        // `EpochMapItem`s, which is the only case when `PartialOrd` is
-        // undefined
-        self.partial_cmp(other).unwrap()
-    }
-}
-
 #[derive(Clone, Debug)]
-/// In-memory implementation of an `EpochMapper`, which allows to map a `UserEpoch`
-/// to an `IncrementalEpoch` used by storages. It basically handles two types of
-/// mappings, depending on how the epoch maps are inserted:
+/// Data structure employed both for in-memory implementation of an `EpochMapper`,
+/// and as a memory cache for the DB-based `EpochMapper` implementation.
+/// The flag `IS_CACHE` is employed to specify whether the data structure is employed
+/// as a cache or as a standalong in-memory `EpochMapper`.
```

Comment:
standalone?

Reply:
Done in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/425/commits/9a606e9c9e8b321fc79145d50371e345d336870a